libs/datamodel/connectors/sql-datamodel-connector/src/mysql_datamodel_connector.rs

```diff
@@ -178,45 +182,45 @@ impl Connector for MySqlDatamodelConnector {
             BIG_INT_TYPE_NAME => MySqlType::BigInt,
             UNSIGNED_BIG_INT_TYPE_NAME => MySqlType::UnsignedBigInt,
             DECIMAL_TYPE_NAME => {
-                if let (Some(first_arg), Some(second_arg)) = (args.get(0), args.get(1)) {
```

Comment:
Does this here mean you can have either two or more parameters without a validation error? At least it looks like that! Also, I'd like to point you to a concept of slice patterns. They look like to be a nicer solution here as least how it looks, but if I'm right about the bug of allowing more than two params here (and the matches afterwards), we'll fix two problems here! See:

```rust
DECIMAL_TYPE_NAME => match parsed_args.as_slice() {
    // Matches two and only two params.
    [scale, precision] => MySqlType::Decimal(*scale, *precision),
    // Matches all other parameter numbers.
    _ => {
        return Err(ConnectorError::new_argument_count_mismatch_error(
            DECIMAL_TYPE_NAME,
            2,
            parsed_args.len(),
        ));
    }
},
```

Reply:
https://doc.rust-lang.org/reference/patterns.html#slice-patterns

Reply:
No, there is a validation before, checking the number of arguments. So the `argument_count_mismatch_error` will never be called actually, I guess this should be removed here? It felt wrong to not error here though 🤷 

(here's the validation before: https://github.com/prisma/prisma-engines/blob/d6e1f3b44115a01f67b4f9cc7123783065db4837/libs/datamodel/core/src/transform/ast_to_dml/lift.rs#L295)

Apart from that, the slice pattern does look cleaner, I'll use that.